tests/Unit/Hooks/HasPsCarrierListHooksTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/** @noinspection AutoloadingIssuesInspection,AutoloadingIssuesInspection,PhpIllegalPsrClassPathInspection,PhpUnused,StaticClosureCanBeUsedInspection,PhpUnhandledExceptionInspection */
+
+declare(strict_types=1);
+
+namespace MyParcelNL\PrestaShop\Hooks;
+
+use Address;
+use Carrier as PsCarrier;
+use Cart;
+use Cookie;
+use Country;
+use MyParcelNL\Pdk\Account\Platform;
+use MyParcelNL\Pdk\Carrier\Model\Carrier;
+use MyParcelNL\PrestaShop\Entity\MyparcelnlCarrierMapping;
+use MyParcelNL\PrestaShop\Tests\Uses\UsesMockPsPdkInstance;
+use MyParcelNL\Sdk\src\Support\Arr;
+use function MyParcelNL\Pdk\Tests\mockPlatform;
+use function MyParcelNL\Pdk\Tests\usesShared;
+use function MyParcelNL\PrestaShop\psFactory;
+
+usesShared(new UsesMockPsPdkInstance());
+
+class WithHasPsCarrierListHooks
+{
+    use HasPsCarrierListHooks;
+}
+
+it('filters carriers from delivery options list', function (
+    string $isoCode,
+    array  $filteredCarriers,
+    string $platform = Platform::MYPARCEL_NAME
+) {
+    $reset = mockPlatform($platform);
+
+    $allCarriers = [
+        Carrier::CARRIER_BPOST_NAME,
+        Carrier::CARRIER_DHL_FOR_YOU_NAME,
+        sprintf('%s:1234', Carrier::CARRIER_DHL_FOR_YOU_NAME),
+        Carrier::CARRIER_DHL_PARCEL_CONNECT_NAME,
+        Carrier::CARRIER_DPD_NAME,
+        Carrier::CARRIER_POSTNL_NAME,
+        Carrier::CARRIER_UPS_NAME,
+    ];
+
+    $deliveryOptionCarrierList = [
+        '22,' => [
```

Comment:
Waarom begint de index niet bij 0?

Reply:
zo ziet het er ook in de praktijk ongeveer uit